// @webstudio-is/react-sdk/lib/css/normalize.js
// src/css/presets.ts
var borders = [
  {
    property: "borderTopWidth",
    value: { type: "unit", value: 1, unit: "px" }
  },
  {
    property: "borderRightWidth",
    value: { type: "unit", value: 1, unit: "px" }
  },
  {
    property: "borderBottomWidth",
    value: { type: "unit", value: 1, unit: "px" }
  },
  {
    property: "borderLeftWidth",
    value: { type: "unit", value: 1, unit: "px" }
  }
];
var outline = [
  {
    property: "outlineWidth",
    value: { type: "unit", value: 1, unit: "px" }
  }
];

// src/css/normalize.ts
var boxSizing = {
  property: "boxSizing",
  value: { type: "keyword", value: "border-box" }
};
var baseStyle = [boxSizing, ...borders, ...outline];
var div = baseStyle;
var address = baseStyle;
var article = baseStyle;
var aside = baseStyle;
var figure = baseStyle;
var footer = baseStyle;
var header = baseStyle;
var main = baseStyle;
var nav = baseStyle;
var section = baseStyle;
var form = baseStyle;
var label = baseStyle;
var h1 = baseStyle;
var h2 = baseStyle;
var h3 = baseStyle;
var h4 = baseStyle;
var h5 = baseStyle;
var h6 = baseStyle;
var i = baseStyle;
var img = baseStyle;
var a = baseStyle;
var li = baseStyle;
var ul = baseStyle;
var ol = baseStyle;
var p = baseStyle;
var span = baseStyle;
var html = [
  /* 1 */
  {
    property: "lineHeight",
    value: { type: "unit", value: 1.15, unit: "number" }
  },
  /* 2 */
  {
    property: "textSizeAdjust",
    value: { type: "unit", value: 100, unit: "%" }
  },
  /* 3 */
  {
    property: "tabSize",
    value: { type: "unit", value: 4, unit: "number" }
  },
  boxSizing,
  ...borders
];
var body = [
  /* 1 */
  {
    property: "marginTop",
    value: { type: "unit", value: 0, unit: "number" }
  },
  {
    property: "marginRight",
    value: { type: "unit", value: 0, unit: "number" }
  },
  {
    property: "marginBottom",
    value: { type: "unit", value: 0, unit: "number" }
  },
  {
    property: "marginLeft",
    value: { type: "unit", value: 0, unit: "number" }
  },
  /* 2 */
  {
    property: "fontFamily",
    value: {
      type: "keyword",
      value: "Arial, Roboto, sans-serif"
    }
  },
  {
    property: "fontSize",
    value: { type: "unit", unit: "px", value: 16 }
  },
  {
    property: "lineHeight",
    value: { type: "unit", unit: "number", value: 1.2 }
  },
  boxSizing,
  ...borders
];
var hr = [
  /* 1 */
  {
    property: "height",
    value: { type: "unit", value: 0, unit: "number" }
  },
  /* 2 */
  {
    property: "color",
    value: { type: "keyword", value: "inherit" }
  },
  boxSizing,
  ...borders
];
var b = [
  {
    property: "fontWeight",
    value: { type: "keyword", value: "700" }
  },
  boxSizing,
  ...borders
];
var strong = b;
var code = [
  /* 1 */
  {
    property: "fontFamily",
    value: {
      type: "keyword",
      value: `ui-monospace, SFMono-Regular, Consolas, 'Liberation Mono', Menlo, monospace`
    }
  },
  /* 2 */
  {
    property: "fontSize",
    value: { type: "unit", value: 1, unit: "em" }
  },
  boxSizing,
  ...borders
];
var kbd = code;
var samp = code;
var pre = code;
var small = [
  {
    property: "fontSize",
    value: { type: "unit", value: 80, unit: "%" }
  },
  boxSizing,
  ...borders
];
var subSupBase = [
  {
    property: "fontSize",
    value: { type: "unit", value: 75, unit: "%" }
  },
  {
    property: "lineHeight",
    value: { type: "unit", value: 0, unit: "number" }
  },
  {
    property: "position",
    value: { type: "keyword", value: "relative" }
  },
  {
    property: "verticalAlign",
    value: { type: "keyword", value: "baseline" }
  },
  boxSizing,
  ...borders
];
var sub = [
  ...subSupBase,
  {
    property: "bottom",
    value: { type: "unit", value: -0.25, unit: "em" }
  }
];
var sup = [
  ...subSupBase,
  {
    property: "top",
    value: { type: "unit", value: -0.5, unit: "em" }
  }
];
var table = [
  /* 1 */
  {
    property: "textIndent",
    value: { type: "unit", value: 0, unit: "number" }
  },
  ...borders,
  /* 2 */
  {
    property: "borderTopColor",
    value: { type: "keyword", value: "inherit" }
  },
  {
    property: "borderRightColor",
    value: { type: "keyword", value: "inherit" }
  },
  {
    property: "borderBottomColor",
    value: { type: "keyword", value: "inherit" }
  },
  {
    property: "borderLeftColor",
    value: { type: "keyword", value: "inherit" }
  },
  boxSizing
];
var buttonBase = [
  /* 1 */
  {
    property: "fontFamily",
    value: { type: "keyword", value: "inherit" }
  },
  {
    property: "fontSize",
    value: { type: "unit", value: 100, unit: "%" }
  },
  {
    property: "lineHeight",
    value: { type: "unit", value: 1.15, unit: "number" }
  },
  /* 2 */
  {
    property: "marginTop",
    value: { type: "unit", value: 0, unit: "number" }
  },
  {
    property: "marginRight",
    value: { type: "unit", value: 0, unit: "number" }
  },
  {
    property: "marginBottom",
    value: { type: "unit", value: 0, unit: "number" }
  },
  {
    property: "marginLeft",
    value: { type: "unit", value: 0, unit: "number" }
  },
  boxSizing,
  ...borders
];
var input = buttonBase;
var optgroup = buttonBase;
var textarea = buttonBase;
var button = [
  ...buttonBase,
  {
    property: "textTransform",
    value: { type: "keyword", value: "none" }
  }
];
var select = button;
var legend = [
  {
    property: "paddingTop",
    value: { type: "unit", value: 0, unit: "number" }
  },
  {
    property: "paddingRight",
    value: { type: "unit", value: 0, unit: "number" }
  },
  {
    property: "paddingBottom",
    value: { type: "unit", value: 0, unit: "number" }
  },
  {
    property: "paddingLeft",
    value: { type: "unit", value: 0, unit: "number" }
  },
  boxSizing,
  ...borders
];
var progress = [
  {
    property: "verticalAlign",
    value: { type: "keyword", value: "baseline" }
  },
  boxSizing,
  ...borders
];
var summary = [
  {
    property: "display",
    value: { type: "keyword", value: "list-item" }
  },
  boxSizing,
  ...borders
];
export {
  a,
  address,
  article,
  aside,
  b,
  body,
  button,
  code,
  div,
  figure,
  footer,
  form,
  h1,
  h2,
  h3,
  h4,
  h5,
  h6,
  header,
  hr,
  html,
  i,
  img,
  input,
  kbd,
  label,
  legend,
  li,
  main,
  nav,
  ol,
  optgroup,
  p,
  pre,
  progress,
  samp,
  section,
  select,
  small,
  span,
  strong,
  sub,
  summary,
  sup,
  table,
  textarea,
  ul
};
