// @webstudio-is/image/lib/index.js
// src/image.tsx
import { forwardRef } from "react";

// src/image-optimize.ts
var imageSizes = [16, 32, 48, 64, 96, 128, 256, 384];
var deviceSizes = [640, 750, 828, 1080, 1200, 1920, 2048, 3840];
var allSizes = [...imageSizes, ...deviceSizes];
var getWidths = (width, sizes) => {
  if (sizes) {
    const viewportWidthRe = /(^|\s)(1?\d?\d)vw/g;
    const percentSizes = [];
    for (let match; match = viewportWidthRe.exec(sizes); match) {
      percentSizes.push(Number.parseInt(match[2], 10));
    }
    if (percentSizes.length) {
      const smallestRatio = Math.min(...percentSizes) * 0.01;
      return {
        widths: allSizes.filter(
          (size) => size >= deviceSizes[0] * smallestRatio
        ),
        kind: "w"
      };
    }
    return { widths: allSizes, kind: "w" };
  }
  if (width == null) {
    return { widths: deviceSizes, kind: "w" };
  }
  const MAX_DEVICE_PIXEL_RATIO = 2;
  let index = allSizes.findIndex(
    (size) => size >= MAX_DEVICE_PIXEL_RATIO * width
  );
  index = index < 0 ? allSizes.length : index;
  return {
    widths: allSizes.slice(0, index + 1),
    kind: "w"
  };
};
var generateImgAttrs = ({
  src,
  width,
  quality,
  sizes,
  loader
}) => {
  const { widths, kind } = getWidths(width, sizes);
  return {
    sizes: !sizes && kind === "w" ? "100vw" : sizes,
    srcSet: widths.map(
      (w, i) => `${loader({ src, quality, width: w })} ${kind === "w" ? w : i + 1}${kind}`
    ).join(", "),
    // Must be last, to prevent Safari to load images twice
    src: loader({
      src,
      quality,
      width: widths[widths.length - 1]
    })
  };
};
var getInt = (value) => {
  if (typeof value === "number") {
    return Math.round(value);
  }
  if (typeof value === "string") {
    const vNum = Number.parseFloat(value);
    if (!Number.isNaN(vNum)) {
      return Math.round(vNum);
    }
  }
  return void 0;
};
var DEFAULT_SIZES = "(min-width: 1280px) 50vw, 100vw";
var DEFAULT_QUALITY = 80;
var getImageAttributes = (props) => {
  const width = getInt(props.width);
  const quality = Math.max(
    Math.min(getInt(props.quality) ?? DEFAULT_QUALITY, 100),
    0
  );
  if (props.src != null && props.src !== "") {
    if (props.srcSet == null && props.optimize) {
      const sizes = props.sizes ?? (props.width == null ? DEFAULT_SIZES : void 0);
      return generateImgAttrs({
        src: props.src,
        width,
        quality,
        sizes,
        loader: props.loader
      });
    }
    const resAttrs = { src: props.src };
    if (props.srcSet != null) {
      resAttrs.srcSet = props.srcSet;
    }
    if (props.sizes != null) {
      resAttrs.sizes = props.sizes;
    }
    return resAttrs;
  }
  return null;
};

// src/image.tsx
import { jsx } from "react/jsx-runtime";
var Image = forwardRef(
  ({
    quality,
    loader,
    optimize = true,
    loading = "lazy",
    decoding = "async",
    ...imageProps
  }, ref) => {
    const imageAttributes = getImageAttributes({
      src: imageProps.src,
      srcSet: imageProps.srcSet,
      sizes: imageProps.sizes,
      width: imageProps.width,
      quality,
      loader,
      optimize
    }) ?? { src: imagePlaceholderSvg };
    return /* @__PURE__ */ jsx(
      "img",
      {
        ...imageProps,
        ...imageAttributes,
        decoding,
        loading,
        ref
      }
    );
  }
);
Image.displayName = "Image";
var imagePlaceholderSvg = `data:image/svg+xml;base64,${btoa(`<svg
  width="140"
  height="140"
  viewBox="0 0 600 600"
  fill="none"
  xmlns="http://www.w3.org/2000/svg"
  >
  <rect width="600" height="600" fill="#CCCCCC" />
  <path
    fill-rule="evenodd"
    clip-rule="evenodd"
    d="M450 170H150C141.716 170 135 176.716 135 185V415C135 423.284 141.716 430 150 430H450C458.284 430 465 423.284 465 415V185C465 176.716 458.284 170 450 170ZM150 145C127.909 145 110 162.909 110 185V415C110 437.091 127.909 455 150 455H450C472.091 455 490 437.091 490 415V185C490 162.909 472.091 145 450 145H150Z"
    fill="#A2A2A2"
  />
  <path
    d="M237.135 235.012C237.135 255.723 220.345 272.512 199.635 272.512C178.924 272.512 162.135 255.723 162.135 235.012C162.135 214.301 178.924 197.512 199.635 197.512C220.345 197.512 237.135 214.301 237.135 235.012Z"
    fill="#A2A2A2"
  />
  <path
    d="M160 405V367.205L221.609 306.364L256.552 338.628L358.161 234L440 316.043V405H160Z"
    fill="#A2A2A2"
  />
</svg>`)}`;

// src/image-loaders.ts
import warnOnce from "warn-once";
var createImageLoader = (loaderOptions) => ({ width, src, quality, format }) => {
  if (true) {
    warnOnce(
      allSizes.includes(width) === false,
      "Width must be only from allowed values"
    );
  }
  const { imageBaseUrl } = loaderOptions;
  const searchParams = new URLSearchParams();
  searchParams.set("width", width.toString());
  searchParams.set("quality", quality.toString());
  searchParams.set("format", format ?? "auto");
  return `${imageBaseUrl}${src}?${searchParams.toString()}`;
};
export {
  Image,
  allSizes,
  createImageLoader,
  getImageAttributes
};
